### PR TITLE
Accept raw BPF filters, configurable LinkType

### PR DESCRIPTION
**Pull request by arthurfabre**
* Accept raw cBPF instructions
  
  Extend the cli / flag parsing to accept raw cBPF instructions, in the
  form typically used by BPF tools (bpf_asm, bpf_dbg..). This allows other
  helpers that emit cBPF directly to be used, for example to match
  encapsulated traffic.

* Configurable LinkType
  
  The underlying network / physical layer exposed to xdpcap is not
  necessarily ethernet. Unfortunately we don't know what interface we're
  actually attached to, so we can't auto detect the correct link type.
    
  Add a configurable flag for the link type, defaulting to ethernet.

* Update newtools, cbpfc
  
  Update our dependencies, including newtools and cbpfc. This pulls in
  support for the cBPF length extension.

### CODE REVIEW

File: cmd/xdpcap/flags.go [hunk 1]
```diff
@@ -5,9 +5,13 @@ import (
 	"fmt"
 	"io/ioutil"
 	"os"
+	"regexp"
+	"strconv"
 	"strings"
 
+	"github.com/google/gopacket/layers"
```
Comment (lmb) [inline cmd/xdpcap/flags.go]: Is this already a dependency? If not, I think it's worth avoiding adding this. gopacket is huge.

File: cmd/xdpcap/flags.go [hunk 2]
```diff
@@ -28,15 +32,136 @@ func (a *actionsFlag) Set(val string) error {
...
+		if err != nil {
+			return nil, errors.Wrapf(err, "instruction %d", i)
+		}
+	}
+
+	return cbpfInsns, nil
+}
+
+// parseInstruction parses a string of cBPF 4 tuple instructions, formatted as:
+// <length>,<opcode> <jt> <jf> <k>,...
```
Comment (lmb) [inline cmd/xdpcap/flags.go]: It doesn't parse length though?

File: cmd/xdpcap/flags.go (hunk 1 above)
Comment (lmb) [reply cmd/xdpcap/flags.go]: Maybe some of the constants from https://godoc.org/golang.org/x/sys/unix would do?

Comment (arthurfabre) [reply cmd/xdpcap/flags.go]: Unfortunately we already have to pull in `gopacket/layers` to use it with `gopacket/pcap` (the `libpcap` interface).

File: cmd/xdpcap/flags.go (hunk 2 above)
Comment (arthurfabre) [reply cmd/xdpcap/flags.go]: Oops yes, that comment is wrong. Updated :)

**Pull request merged**